src/summary.rs

```diff
@@ -89,34 +96,44 @@ impl SummaryFile {
                 };
                 let summary_text =
                     toml::to_string(&initialization).context("Failed to serialize Summary")?;
-                summary_file = OpenOptions::new()
+                let mut summary_file_handle = OpenOptions::new()
                     .read(true)
                     .write(true)
                     .truncate(true)
                     .open(&summary_path)
                     .await
                     .context("couldn't open new summary file")?;
-                summary_file
+                summary_file_handle
                     .write_all(summary_text.as_bytes())
                     .await
                     .context("write to summary failed")?;
-                summary_file.flush().await.context("flush at creation failed")?;
-                summary_file
+                summary_file_handle.flush().await.context("flush at creation failed")?;
+                summary_file_handle
                     .seek(std::io::SeekFrom::Start(0))
                     .await
                     .context("couldn't seek to the beginning of the summary file")?;
 
-                return Ok(SummaryFile { inner: Arc::new(Mutex::new(summary_file)) });
+                Ok(SummaryFile { inner: Arc::new(Mutex::new(summary_file_handle)) })
+            }
+            Ok(summary_file_handle) => {
+                let summary_file = SummaryFile { inner: Arc::new(Mutex::new(summary_file_handle)) };
+                if let Some(user_space_pledged) = maybe_user_space_pledged {
+                    let (summary, _) = summary_file.read_and_deserialize().await?;
+                    summary_file
+                        .update(SummaryUpdateFields {
+                            is_plotting_finished: summary.initial_plotting_finished,
+                            maybe_updated_user_space_pledged: Some(user_space_pledged),
+                            // No change in other values
+                            new_authored_count: 0,
+                            new_vote_count: 0,
+                            new_reward: Rewards(0),
+                            new_parsed_blocks: 0,
+                        })
```

Comment:
When instantiating structs please order fields in the same exact order as in struct definition, it is confusing otherwise.